File: springboot-backend/src/main/java/com/capstone/project/service/CommentService.java
package com.capstone.project.service;
import com.capstone.project.exception.ResourceNotFroundException;
import com.capstone.project.model.Comment;

import java.text.ParseException;
import java.util.List;
import java.util.Map;

public interface CommentService {

    List<Comment> getAllComment();
    List<Comment> getAllCommentByPostId(int id);
    List<Comment> getAllCommentByRootId(int id);
    List<Comment> getAllCommentByStudySetId(int id);
    List<Comment> getAllCommentByTestId(int id);

    Comment getCommentById(int id) throws ResourceNotFroundException;

    Comment createComment(Comment comment);

    Comment updateComment( Comment comment,  int id) throws ResourceNotFroundException;

    Boolean deleteComment (int id) throws ResourceNotFroundException;

    Map<String, Object> getFilterComment( String search, String author, int typeid,int postid,int testid,int studysetid,int rootid, int page, int size) throws ResourceNotFroundException;
}

File: react-frontend/src/services/CommentService.js
import axios from 'axios'

const API_BASE_URL = process.env.REACT_APP_HOST

const getCommentById = (id) => {
    return axios.get(API_BASE_URL + '/comments/' + id)
}

const getFilterList = (
    search,
    author,
    direction,
    typeid,
    postid,
    rootId,
    page,
    size
) => {
    return axios.get(
        API_BASE_URL +
            '/filtercomment?search' +
            search +
            '&author' +
            author +
            '&direction' +
            direction +
            '&typeid' +
            typeid +
            '&postid' +
            postid +
            '&rootid' +
            rootId +
            '&page' +
            page +
            '&size' +
            size
    )
}

const getAllCommentByPostId = (id) => {
    return axios.get(API_BASE_URL + '/commentbypostid/' + id)
}

const getAllCommentByAssignmentId = (id) => {
    return axios.get(API_BASE_URL + '/commentbasignmentid/' + id)
}

const getAllCommentBySubmisionId = (id) => {
    return axios.get(API_BASE_URL + '/commentbysubmisionid/' + id)
}

const getAllCommentByTestId = (id) => {
    return axios.get(API_BASE_URL + '/commentbytestid/' + id)
}

const createComment = (comment) => {
    return axios.post(API_BASE_URL + '/comments', comment)
}

const updateComment = (id, comment) => {
    return axios.put(API_BASE_URL + '/comments/' + id, comment)
}

const deleteComment = (id) => {
    return axios.delete(API_BASE_URL + '/comments/' + id)
}

const CommentService = {
    getCommentById,
    getFilterList,
    getAllCommentByPostId,
    createComment,
    updateComment,
    deleteComment,
    getAllCommentByAssignmentId,
    getAllCommentBySubmisionId,
    getAllCommentByTestId,
}

export default CommentService
